src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -174,38 +168,49 @@ def q_impl(
         alpha = self.attrs.get("alpha", 1)
         beta = self.attrs.get("beta", 1)
 
+        # If self.constant_inputs is empty this is an encrypted gemm
+        is_encrypted_gemm = isinstance(self.constant_inputs, dict) and not self.constant_inputs
```

Comment:
there's is no better way to know this ? this does not seem very robust honestly

Reply:
Not currently. We would need a bigger PR that propagate the encryption status. I agree, this might be a problem if one of the input is in clear. cc @fd0r

Reply:
Let me add an issue here.

Reply:
ok got it thanks for the issue !